src/compiler/commandLineParser.ts

```diff
@@ -2169,6 +2170,17 @@ namespace ts {
                 errors.push(createCompilerDiagnostic(message, arg0, arg1));
             }
         }
+
+        function makeFilesReferencesAbsolute(optionsFromCLI: CompilerOptions) {
+            Object.keys(optionsFromCLI).forEach(key => {

```

Comment:
I don't think existing options should be relative to config File. Infact they should be relative to host.currentDirectory since I might be parsing using `tsc -p src\tsconfig.json --outDir bin` and here it should mean bin is relative to currentDirectory and not `src\bin`

Reply:
Also I think this should happen in `tsc.ts` instead of parseCommandLineWorker. API user should handle what existingOptions are..

Reply:
👍  thanks!